extraConfigDpuInfra.py

```diff
@@ -186,24 +187,36 @@ def cb() -> None:
     client.oc("create -f manifests/infra/dpuclusterconfig.yaml")
 
     logger.info("Patching mcp setting maxUnavailable to 2")
-    client.oc("patch mcp dpu --type=json -p=\\[\\{\"op\":\"replace\",\"path\":\"/spec/maxUnavailable\",\"value\":2\\}\\]")
+    arg = [
+        {
+            "op": "replace",
+            "path": "/spec/maxUnavailable",
+            "value": 2,
+        }
+    ]
+    client.oc(f"patch mcp dpu --type=json -p={json.dumps(arg)}")
 
     logger.info("Labeling nodes")
     for b in bf_names:
         client.oc(f"label node {b} node-role.kubernetes.io/dpu-worker=")
 
-    logger.info("Creating config map")
-    logger.info(client.oc("create -f manifests/infra/cm.yaml"))
+    if not new_api:
+        # DELTA: No need to create config map to set dpu mode. https://github.com/openshift/cluster-network-operator/pull/1676
+        logger.info("Creating config map")
+        logger.info(client.oc("create -f manifests/infra/cm.yaml"))
 
     for b in bf_names:
         client.oc(f"label node {b} network.operator.openshift.io/dpu=")
     logger.info("Waiting for mcp to be ready")
-    client.wait_for_mcp("dpu", "cm.yaml")
+    if new_api:
+        client.wait_for_mcp("dpu", "dpu mcp patch")
```

Comment:
This is just a string that provides logging, no need to have an if statement to change this based on API version. Find to just use cm.yaml for each.